include/deal.II/matrix_free/vector_data_exchange.h

```diff
@@ -0,0 +1,255 @@
+
+        virtual void
+        import_from_ghosted_array_finish(
+          const VectorOperation::values               vector_operation,
+          const ArrayView<double> &                   locally_owned_storage,
+          const std::vector<ArrayView<const double>> &shared_arrays,
+          const ArrayView<double> &                   ghost_array,
+          const ArrayView<const double> &             temporary_storage,
+          std::vector<MPI_Request> &                  requests) const = 0;
+
+        virtual void
+        reset_ghost_values(const ArrayView<double> &ghost_array) const = 0;
+
+        virtual void
+        export_to_ghosted_array_start(
+          const unsigned int                         communication_channel,
+          const ArrayView<const float> &             locally_owned_array,
+          const std::vector<ArrayView<const float>> &shared_arrays,
+          const ArrayView<float> &                   ghost_array,
+          const ArrayView<float> &                   temporary_storage,
+          std::vector<MPI_Request> &                 requests) const = 0;
+
+        virtual void
+        export_to_ghosted_array_finish(
+          const ArrayView<const float> &             locally_owned_array,
+          const std::vector<ArrayView<const float>> &shared_arrays,
+          const ArrayView<float> &                   ghost_array,
+          std::vector<MPI_Request> &                 requests) const = 0;
+
+        virtual void
+        import_from_ghosted_array_start(
+          const VectorOperation::values              vector_operation,
+          const unsigned int                         communication_channel,
+          const ArrayView<const float> &             locally_owned_array,
+          const std::vector<ArrayView<const float>> &shared_arrays,
+          const ArrayView<float> &                   ghost_array,
+          const ArrayView<float> &                   temporary_storage,
+          std::vector<MPI_Request> &                 requests) const = 0;
+
+        virtual void
+        import_from_ghosted_array_finish(
+          const VectorOperation::values              vector_operation,
+          const ArrayView<float> &                   locally_owned_storage,
+          const std::vector<ArrayView<const float>> &shared_arrays,
+          const ArrayView<float> &                   ghost_array,
+          const ArrayView<const float> &             temporary_storage,
+          std::vector<MPI_Request> &                 requests) const = 0;
+
+        virtual void
+        reset_ghost_values(const ArrayView<float> &ghost_array) const = 0;
+      };
+
+
+      /**
+       * Class that simply delegates the task to a Utilities::MPI::Partitioner.
+       */
+      class PartitionerWrapper : public Base
+      {
+      public:
+        PartitionerWrapper(
+          const std::shared_ptr<const Utilities::MPI::Partitioner> partitioner);
```

Comment:
To make `clang-tidy` happy, you should pass this by const reference.

Reply:
I have made the modification!